projects/bp-gallery/src/components/views/location-curating/LocationManagementDialog.tsx

```diff
@@ -0,0 +1,392 @@
+                        []
+                      }
+                      allTags={(flattenedTags as any) ?? []}
+                      onChange={locations => {
+                        setChildTags(locations as FlatTag[]);
+                      }}
+                      noContentText={''}
+                      fixedParentTag={locationTag}
+                      customChipOnClick={(id: string) => {
+                        parentTagHistory.current.push(parentTag);
+                        setParentTag(locationTag);
+                        setLocationTagID(id);
+                      }}
+                      allowCreateChild={true}
+                    />
+                  </PictureInfoField>
+                </div>
+                <div className='location-management-parents-container'>
+                  <div>{t('common.superlocations')}</div>
+                  <PictureInfoField
+                    title={t('common.superlocations')}
+                    icon={<Place />}
+                    type='location'
+                  >
+                    <TagSelectionField
+                      type={TagType.LOCATION}
+                      tags={
+                        (locationTag.parent_tags?.map(tag => ({
+                          ...tag,
+                          verified: true,
+                        })) as any) ?? []
+                      }
+                      allTags={(flattenedTags as any) ?? []}
+                      onChange={locations => {
+                        setParentTags(locations as FlatTag[]);
+                      }}
+                      noContentText={''}
+                      fixedChildTag={locationTag}
+                      customChipOnClick={(id: string) => {
+                        setParentTag(parentTagHistory.current.pop());
+                        setLocationTagID(id);
+                      }}
+                      createParentMutation={newLocationTagMutation}
+                    />
+                  </PictureInfoField>
+                </div>
+              </div>
+            </div>
+            <div className='location-management-right'>
+              <div className='location-management-map'></div>
+              <div className='location-management-actions'>
+                <div className='location-management-picture-count'>
+                  {flattenedPictures &&
+                    t('tag-panel.location-pictures', {
+                      count: flattenedPictures.locationTag.pictures.length,
+                    })}
+                </div>
+                <Button
+                  className='location-management-show-pictures-button'
+                  onClick={() => {
+                    handleClose(undefined);
+                    history.push(`/show-more/location/${locationTag.id}`, {
+                      showBack: true,
+                    });
+                  }}
+                  endIcon={<ArrowForwardIos />}
+                >
```

Comment:
I think it should be more versatile before we start using it everywhere. I think it should be a separate issue, instead of me adapting it to fit two use cases instead of one.